lib/_http_server.js

```diff
@@ -722,10 +722,12 @@ function onParserExecuteCommon(server, socket, parser, state, ret, d) {
 
       socket.readableFlowing = null;
 
+      socket.unshift(bodyHead);
```

Comment:
```suggestion
      if (bodyHead.length > 0) socket.unshift(bodyHead);
```

I think for consistency the same should be done also on the client (`_http_client.js`).